azkaban-common/src/main/java/azkaban/storage/StorageManager.java

```diff
@@ -71,4 +65,20 @@ public void uploadProject(
         metadata, localFile.getName(), localFile.length()));
     storage.put(metadata, localFile);
   }
+
+  /**
+   * Fetch project file from storage.
+   *
+   * @param projectId required project ID
+   * @param version version to be fetched
+   * @return Handler object containing hooks to fetched project file
+   */
+  public ProjectFileHandler getProjectFile(final int projectId, final int version) {
+    log.info(String.format("Fetching project file. project ID: %d version: %d", projectId, version));
+    // TODO spyne: remove huge hack ! There should not be any special handling for Database Storage.
+    if (storage instanceof DatabaseStorage) {
+      return ((DatabaseStorage) storage).get(projectId, version);
```

Comment:
There is no problem as such. It is just that I would prefer storage key to get a single parameter, `URI` in this case.

For example in LOCAL or HDFS storage, I would get the artifact based on path.